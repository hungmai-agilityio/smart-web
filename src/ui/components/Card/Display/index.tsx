// Interfaces
import { IProject } from '@/interface';

// Components
import { CardImage, StatsRow } from '@/ui/components';

interface CardDisplayProps {
  item: IProject;
}

export const CardDisplay = ({ item }: CardDisplayProps) => {
  return (
    <div className="border rounded-lg shadow-md">
      <div className="w-full aspect-image relative">
        <CardImage src={item.image} alt={item.name} autoSize isBlur>
          <StatsRow
            views={item.views}
            favorite={item.favorite}
            comment={item.comment}
            viewIcon="/view-white.svg"
            favoriteIcon="/likes-white.svg"
            commentIcon="/cmt-white.svg"
            isColumn
          />
        </CardImage>
      </div>

      <div className="md:flex hidden items-center w-full justify-between py-3 text-x-xs">
        <StatsRow
          views={item.views}
          favorite={item.favorite}
          comment={item.comment}
          viewIcon="/view-dark.svg"
          favoriteIcon="/likes.svg"
          commentIcon="/comments.svg"
        />
      </div>
    </div>
  );
};
