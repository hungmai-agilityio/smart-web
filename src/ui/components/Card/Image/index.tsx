'use client';
import { ReactNode, useState } from 'react';
import { clsx } from 'clsx';
import Image, { ImageProps } from 'next/image';

interface CardImageProps extends ImageProps {
  children?: ReactNode;
  styles?: string;
  isCircle?: boolean;
  fixedSize?: { width: number; height: number };
  autoSize?: boolean;
  isBlur?: boolean;
}

export const CardImage = ({
  src,
  alt,
  children,
  styles,
  isCircle,
  fixedSize,
  autoSize,
  isBlur,
  ...res
}: CardImageProps) => {
  const [hovered, setHovered] = useState(false);

  const toggleHover = () => setHovered(!hovered);

  const containerStyles = clsx('relative', styles, {
    'rounded-full overflow-hidden border-4 border-gray-300 hover:border-second':
      isCircle,
    'w-full h-full': autoSize,
    [`w-[${fixedSize?.width}px] h-[${fixedSize?.height}px]`]:
      fixedSize && !autoSize
  });

  return (
    <div
      className={containerStyles}
      onMouseEnter={toggleHover}
      onMouseLeave={toggleHover}
    >
      <Image
        src={src}
        alt={alt}
        fill={autoSize}
        width={fixedSize?.width}
        height={fixedSize?.height}
        style={{ objectFit: 'cover' }}
        className={clsx('transition-opacity duration-300', {
          'opacity-50': hovered && children
        })}
        placeholder={isBlur ? 'blur' : 'empty'}
        blurDataURL={
          isBlur
            ? 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mPkNA3fBAACLQFJQfEBVAAAAABJRU5ErkJggg=='
            : undefined
        }
        {...res}
      />
      {hovered && children && (
        <div className="absolute inset-0 flex items-center bg-black bg-opacity-50 text-white p-4">
          {children}
        </div>
      )}
    </div>
  );
};
