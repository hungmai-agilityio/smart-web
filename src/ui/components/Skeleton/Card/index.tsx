export const CardSkeleton = () => {
  return (
    <div className="grid md:grid-cols-4 grid-cols-1 w-5/6 gap-6 mx-auto">
      {Array.from({ length: 4 }).map((_, index) => (
        <div key={index} className="flex flex-col animate-pulse">
          <div className="xl:w-pro-lg xl:h-pro-lg md:w-pro-md md:h-pro-md w-pro-sm h-pro-sm mx-auto bg-gray-300 rounded-full" />

          <div className="mt-4 md:h-32 text-center">
            <div className="h-6 bg-gray-300 rounded w-3/4 mx-auto mb-2"></div>
            <div className="h-4 bg-gray-300 rounded w-1/4 mx-auto mb-2"></div>
            <div className="h-4 bg-gray-300 rounded w-2/4 mx-auto"></div>
          </div>

          <div className="mt-5 mx-auto bg-gray-300 rounded w-24 h-8"></div>
        </div>
      ))}
    </div>
  );
};
