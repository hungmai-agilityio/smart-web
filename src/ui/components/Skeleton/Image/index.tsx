interface SkeletonProps {
  col: string;
  row: number;
}

export const SkeletonImage = ({ col, row }: SkeletonProps) => (
  <div className={`grid md:grid-cols-${col} grid-cols-2 gap-6 w-4/6 mx-auto`}>
    {Array.from({ length: row }).map((_, index) => (
      <div key={index} className="animate-pulse flex flex-col sm:flex-row">
        <div className="bg-gray-300 lg:h-place-lg md:h-place-md h-place-sm w-full rounded-md mb-4 sm:mb-0" />
        <div className="flex flex-col justify-end flex-1 sm:ml-10"></div>
      </div>
    ))}
  </div>
);
