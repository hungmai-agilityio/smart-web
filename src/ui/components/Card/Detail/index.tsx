// Interfaces
import { IProduct } from '@/interface';

// Constants
import { montserrat, TYPE } from '@/constants';

// Components
import { Button, CardImage } from '@/ui/components';
import Image from 'next/image';

interface CardDetailProps {
  item: IProduct;
  nameButton?: string;
  onClick?: () => void;
}

export const CardDetail = ({ item, nameButton, onClick }: CardDetailProps) => {
  return (
    <div className="flex flex-col">
      <div className="xl:w-pro-lg xl:h-pro-lg md:w-pro-md md:h-pro-md w-pro-sm h-pro-sm mx-auto">
        <CardImage src={item.image} alt={item.name} autoSize isCircle isBlur>
          <div className="w-full">
            <Image
              src="/view.svg"
              alt="view icon"
              className="md:block hidden mx-auto"
              width={40}
              height={22}
            />
            <div className="flex justify-center items-center gap-3 md:mt-10 mt-0">
              <Image src="/view.svg" alt="view icon" width={22} height={11} />
              <span className="text-sm">{item.view}</span>
            </div>
          </div>
        </CardImage>
      </div>
      <div className="mt-4 md:h-32 overflow-hidden text-center">
        <p className={`${montserrat.className} font-bold lg:text-lg text-base`}>
          {item.name}
        </p>
        <span className="my-3">by</span>
        <p className="lg:text-base text-sm mt-3">{item.users.join(', ')}</p>
      </div>
      {nameButton && (
        <Button variant={TYPE.SECOND} className="mt-5 mx-auto" onClick={onClick}>
          {nameButton}
        </Button>
      )}
    </div>
  );
};
