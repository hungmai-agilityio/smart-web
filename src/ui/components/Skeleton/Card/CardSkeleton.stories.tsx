import { Meta, StoryObj } from '@storybook/react';
import { CardSkeleton } from '@/ui/components/Skeleton/Card';

const meta: Meta<typeof CardSkeleton> = {
  component: CardSkeleton,

  parameters: {
    nextjs: {
      appDirectory: true
    }
  }
};

export default meta;

type Story = StoryObj<typeof CardSkeleton>;

export const Default: Story = {
  args: {}
};
