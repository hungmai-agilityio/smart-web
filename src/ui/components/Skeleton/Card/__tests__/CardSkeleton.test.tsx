import { CardSkeleton } from '@/ui/components/Skeleton/Card';
import { render } from '@testing-library/react';

it('should be render match to snapshot', () => {
  const { container } = render(<CardSkeleton />);
  expect(container).toMatchSnapshot();
});
