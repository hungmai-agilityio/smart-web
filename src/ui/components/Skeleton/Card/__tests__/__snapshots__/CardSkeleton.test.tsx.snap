// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be render match to snapshot 1`] = `
<div>
  <div
    class="grid md:grid-cols-4 grid-cols-1 w-5/6 gap-6 mx-auto"
  >
    <div
      class="flex flex-col animate-pulse"
    >
      <div
        class="xl:w-pro-lg xl:h-pro-lg md:w-pro-md md:h-pro-md w-pro-sm h-pro-sm mx-auto bg-gray-300 rounded-full"
      />
      <div
        class="mt-4 md:h-32 text-center"
      >
        <div
          class="h-6 bg-gray-300 rounded w-3/4 mx-auto mb-2"
        />
        <div
          class="h-4 bg-gray-300 rounded w-1/4 mx-auto mb-2"
        />
        <div
          class="h-4 bg-gray-300 rounded w-2/4 mx-auto"
        />
      </div>
      <div
        class="mt-5 mx-auto bg-gray-300 rounded w-24 h-8"
      />
    </div>
    <div
      class="flex flex-col animate-pulse"
    >
      <div
        class="xl:w-pro-lg xl:h-pro-lg md:w-pro-md md:h-pro-md w-pro-sm h-pro-sm mx-auto bg-gray-300 rounded-full"
      />
      <div
        class="mt-4 md:h-32 text-center"
      >
        <div
          class="h-6 bg-gray-300 rounded w-3/4 mx-auto mb-2"
        />
        <div
          class="h-4 bg-gray-300 rounded w-1/4 mx-auto mb-2"
        />
        <div
          class="h-4 bg-gray-300 rounded w-2/4 mx-auto"
        />
      </div>
      <div
        class="mt-5 mx-auto bg-gray-300 rounded w-24 h-8"
      />
    </div>
    <div
      class="flex flex-col animate-pulse"
    >
      <div
        class="xl:w-pro-lg xl:h-pro-lg md:w-pro-md md:h-pro-md w-pro-sm h-pro-sm mx-auto bg-gray-300 rounded-full"
      />
      <div
        class="mt-4 md:h-32 text-center"
      >
        <div
          class="h-6 bg-gray-300 rounded w-3/4 mx-auto mb-2"
        />
        <div
          class="h-4 bg-gray-300 rounded w-1/4 mx-auto mb-2"
        />
        <div
          class="h-4 bg-gray-300 rounded w-2/4 mx-auto"
        />
      </div>
      <div
        class="mt-5 mx-auto bg-gray-300 rounded w-24 h-8"
      />
    </div>
    <div
      class="flex flex-col animate-pulse"
    >
      <div
        class="xl:w-pro-lg xl:h-pro-lg md:w-pro-md md:h-pro-md w-pro-sm h-pro-sm mx-auto bg-gray-300 rounded-full"
      />
      <div
        class="mt-4 md:h-32 text-center"
      >
        <div
          class="h-6 bg-gray-300 rounded w-3/4 mx-auto mb-2"
        />
        <div
          class="h-4 bg-gray-300 rounded w-1/4 mx-auto mb-2"
        />
        <div
          class="h-4 bg-gray-300 rounded w-2/4 mx-auto"
        />
      </div>
      <div
        class="mt-5 mx-auto bg-gray-300 rounded w-24 h-8"
      />
    </div>
  </div>
</div>
`;
